### PR TITLE
remove CoinStore's dependency on FullBlock

### PR DESCRIPTION
**Pull request by arvidn**
Just pass in the parts of the block necessary to add the block. Even though this is a net increase in lines of code (for now), it improves encapsulation of the CoinStore class and enables simpler tests of it. My main motivation is that it simplifies writing a benchmark for the `CoinStore` class, since I don't have to construct full blocks.

Here's Koan from Tony Van Eerd to further justify this:

> Master, I am honoured by your visit.
> I was out walking my dog. Your function, why does it take BigCommonStruct instead of just x and y?
> BCS is where we keep x and y, master.
> Bark
> He's hungry.
> Shall I prepare him food?
> No, just open the fridge, let him take what he wants.
> #CppKoan

### CODE REVIEW

File: chia/full_node/coin_store.py [hunk 1]
```diff
@@ -61,45 +60,48 @@ async def create(cls, db_wrapper: DBWrapper, cache_size: uint32 = uint32(60000))
         self.coin_record_cache = LRUCache(cache_size)
         return self
 
-    async def new_block(self, block: FullBlock, tx_additions: List[Coin], tx_removals: List[bytes32]):
+    async def new_block(
+        self,
+        height: uint32,
+        timestamp: uint64,
+        included_reward_coins: Set[Coin],
```
Comment (arvidn) [inline chia/full_node/coin_store.py:67]: this is the key change. Instead of taking a `FullBlock` object, take the 3 parts of it that we need. height, timestamp and the reward coins

**Review by mariano54 — APPROVED** [review]
It looks safe and correct to me.

**Comment by arvidn** [comment]
I'm working on a benchmark for the `CoinStore` class, where I want to run various workloads of mostly inserting new coins and mostly marking coins as spent, and possibly a few other operations (but `new_block()` is the most performance critical I think).

Doing that is a lot simpler if I don't have to actually create blocks, I just want to generate additions and removals as fast as possible, all the time should ideally be spent applying those updates to the database.

Looking at the test for `CoinStore`, it's also quite heavy weight, a lot more so than it would need to be. It currently actually generates a chain of a few blocks to run through. With this change, I think the test could be improved to:

* run faster
* be easier to read (and be confident all edge cases are covered)

I was tempted to start adding more, but I refrained for now, to stay on topic. But I think it would be good to do in the future.

**Review by jack60612 — APPROVED** [review]
Looks great to me!
